src/RamIndexAnalysis.h

```diff
@@ -34,16 +34,160 @@
 #include <utility>
 #include <vector>
 
-#define RIA_NIL 0
-#define RIA_INF -1
-
 // define if enable unit tests
 #define M_UNIT_TEST
 
 namespace souffle {
 
 class RamTranslationUnit;
 
+/** search signature of a RAM operation; each bit represents an attribute of a relation.
+ * A one represents that the attribute has an assigned value; a zero represents that
+ * no value exists (i.e. attribute is unbounded) in the search. */
+class SearchSignature {
+public:
+    explicit SearchSignature(size_t arity, uint64_t mask) : bits(arity, false) {
+        size_t len = bits.size();
+        for (size_t i = 0; i < len; ++i) {
+            bits[i] = (mask % 2);
+            mask /= 2;
+        }
+    }
+    // array subscript operator
+    inline bool operator[](std::size_t pos) const {
```

Comment:
Remove all these `inline`s. This qualifier has no effect on template function definitions nor member function definitions within an aggregate definition.

Reply:
I have everything in a header so I used inline to get around ODR. Should I move it to a proper .cpp source file or just keep the inline?

Reply:
Super late, but might as well answer this: When you define a member func inside a class-def it's implicitly `inline`d, there's no need to explicitly say `inline` again. If you defined it in the header, but outside of the class-def, then you'd need `inline` on the definition. e.g.
```
class A { void foo() { /* okay, implicitly `inline`d b/c I'm defined inside a class-def */ } };
class B { void bar(); };
inline void B::bar() { /* explicit `inline` because I'm not defined inside a class, only declared */ }
```